drivers/mipi_dbi/mipi_dbi_bflb.c

```diff
@@ -0,0 +1,773 @@
+	tmp |= DBI_CR_DBI_CS_STRETCH;
+
+	sys_write32(tmp, cfg->base + DBI_CONFIG_OFFSET);
+
+	ret = mipi_dbi_bflb_configure_freqs(dev, config);
+	if (ret != 0) {
+		return ret;
+	}
+
+	/* TODO mode B pixel format handling */
+	tmp = sys_read32(cfg->base + DBI_PIX_CNT_OFFSET);
+	/* 4 bytes per pixel */
+	tmp |= DBI_CR_DBI_PIX_FORMAT;
+	sys_write32(tmp, cfg->base + DBI_PIX_CNT_OFFSET);
+
+	tmp = sys_read32(cfg->base + DBI_FIFO_CONFIG_0_OFFSET);
+	tmp &= ~DBI_FIFO_FORMAT_MASK;
+	if ((config->config.operation & SPI_TRANSFER_LSB) != 0) {
+		tmp |= DBI_FIFO_FORMAT_RGB << DBI_FIFO_FORMAT_SHIFT;
+	} else {
+		tmp |= DBI_FIFO_FORMAT_BGR << DBI_FIFO_FORMAT_SHIFT;
+	}
+	tmp |= DBI_TX_FIFO_CLR;
+	tmp &= ~DBI_FIFO_YUV_MODE;
+	tmp &= ~DBI_DMA_TX_EN;
+	sys_write32(tmp, cfg->base + DBI_FIFO_CONFIG_0_OFFSET);
+
+	tmp = sys_read32(cfg->base + DBI_FIFO_CONFIG_1_OFFSET);
+	tmp &= ~DBI_TX_FIFO_TH_MASK;
+	tmp |= 3U << DBI_TX_FIFO_TH_SHIFT;
+	sys_write32(tmp, cfg->base + DBI_FIFO_CONFIG_1_OFFSET);
+
+	data->configured = *config;
+
+	return 0;
+}
+
+static int mipi_dbi_bflb_command_write(const struct device *dev,
+				     const struct mipi_dbi_config *dbi_config,
+				     uint8_t cmds, const uint8_t *buf, size_t len)
+{
+	const struct mipi_dbi_bflb_config *cfg = dev->config;
+	struct mipi_dbi_bflb_data *data = dev->data;
+	k_timepoint_t end_timeout =
+		sys_timepoint_calc(K_MSEC(DBI_WAIT_TIMEOUT_MS));
+	int ret;
+	volatile uint32_t tmp;
+	uint8_t j;
+	size_t loaded = 0;
+	bool triggered = false;
+
+	if (len > DBI_MAX_CMD_WRITE_LEN) {
+		LOG_ERR("Max command and data write len is 256 bytes");
+		return -EINVAL;
+	}
+
+	ret = k_sem_take(&data->lock, K_FOREVER);
+	if (ret < 0) {
+		return ret;
+	}
+
+	mipi_dbi_bflb_cs_control(dev, dbi_config, true, false);
+
+	ret = mipi_dbi_bflb_configure(dev, dbi_config);
+	if (ret < 0) {
+		goto out;
+	}
+
+	tmp = sys_read32(cfg->base + DBI_CONFIG_OFFSET);
+	tmp &= ~DBI_CR_DBI_DAT_TP;
+	tmp |= DBI_CR_DBI_CMD_EN;
+	tmp |= DBI_CR_DBI_DAT_WR;
+	if (len == 0) {
+		tmp &= ~DBI_CR_DBI_DAT_EN;
+	} else {
+		tmp |= DBI_CR_DBI_DAT_EN;
+		tmp &= ~DBI_CR_DBI_DAT_BC_MASK;
+		tmp |= (len - 1) << DBI_CR_DBI_DAT_BC_SHIFT;
+	}
+	sys_write32(tmp, cfg->base + DBI_CONFIG_OFFSET);
+
+	tmp = sys_read32(cfg->base + DBI_CMD_OFFSET);
+	tmp &= ~DBI_CR_DBI_CMD_MASK;
+	tmp |= cmds << DBI_CR_DBI_CMD_SHIFT;
+	sys_write32(tmp, cfg->base + DBI_CMD_OFFSET);
+
+	mipi_dbi_bflb_clear_fifo(dev);
+
+	tmp = sys_read32(cfg->base + DBI_INT_STS_OFFSET);
+	tmp |= DBI_CR_DBI_END_CLR;
+	sys_write32(tmp, cfg->base + DBI_INT_STS_OFFSET);
+
+	LOG_DBG("cmd write: %x, len: %d", cmds, len);
+
+	if (mipi_dbi_has_pin(&cfg->dc_gpio)) {
+		ret = gpio_pin_set_dt(&cfg->dc_gpio, 1);
+		if (ret < 0) {
+			goto out;
+		}
+	}
+
+	if (len > 0) {
+		while (loaded < len) {
+			tmp = sys_read32(cfg->base + DBI_FIFO_CONFIG_1_OFFSET);
+			if ((tmp & DBI_TX_FIFO_CNT_MASK) > 0) {
+				tmp = 0;
+				j = 0;
+				for (; j < 4 && loaded + j < len; j++) {
+					tmp |= (buf[loaded + j]) << (j * 8);
+				}
```

Comment:
I was thinking: let's use `sys_le32_to_cpu();` but nope(!) your version also handles the edge case when `len % 4 != 0`.

I was thinking: it writes extra 0x00 at the end when `len % 4 != 0` but nope(!) your version setup the length above already (`tmp |= (len - 1) << DBI_CR_DBI_DAT_BC_SHIFT`).

All good :heavy_check_mark: